app/controllers/referral_promos_controller.rb

```diff
@@ -0,0 +1,43 @@
+class ReferralPromosController < ApplicationController
+  before_action :authenticate_publisher!
+
+  def show
```

Comment:
should it be `index`? since we plan to view all active promo codes on one page.

Reply:
Yes `index` feels better suited, I'll change the actions/routes